src/write_gcm.c

```diff
@@ -108,6 +113,10 @@ static const char json_content_type_header[] = "Content-Type: application/json";
 //==============================================================================
 //==============================================================================
 
+static _Bool wg_end_to_end_test_mode() {
+  return atoi(global_option_get("ReadThreads")) == -1;
```

Comment:
How about we go with approach 3 for now and send an email to the collectd list to see what people think about a new global variable?  Otherwise, we'll just need to be extra careful on version upgrades for it.